ionic/ionic.go

```diff
@@ -0,0 +1,110 @@
+package ionic
+
+import (
+	"bufio"
+	"fmt"
+	"os"
+	"regexp"
+	"strings"
+
+	"github.com/bitrise-io/go-utils/command"
+	"github.com/bitrise-io/go-utils/log"
+	ver "github.com/hashicorp/go-version"
+	"github.com/pkg/errors"
+)
+
+// LogIntf declares used logger functions
+type LogIntf interface {
+	// Printf(format string, v ...interface{})
+	Infof(format string, v ...interface{})
+	Donef(format string, v ...interface{})
+}
+
+// Log is the currently set logger
+var Log LogIntf = log.NewDummyLogger()
+
+// Version returns ionic version
+func Version() (*ver.Version, error) {
+	cmd := command.New("ionic", "-v")
+	cmd.SetStdin(strings.NewReader("Y"))
+	out, err := cmd.RunAndReturnTrimmedCombinedOutput()
+	if err != nil {
+		return nil, err
+	}
+
+	// fix for ionic-cli intercative version output: `[1000D[K3.2.0`
+	pattern := `(?P<version>\d+\.\d+\.\d+)`
+
+	reader := strings.NewReader(out)
+	scanner := bufio.NewScanner(reader)
+	for scanner.Scan() {
+		line := scanner.Text()
+		if match := regexp.MustCompile(pattern).FindStringSubmatch(line); len(match) == 2 {
+			versionStr := match[1]
+			version, err := ver.NewVersion(versionStr)
+			if err != nil {
+				return nil, errors.Wrapf(err, "failed to parse version: %s", versionStr)
+			}
+			return version, nil
+		}
+	}
+	if err := scanner.Err(); err != nil {
+		return nil, err
+	}
+
+	return nil, fmt.Errorf("output: %s", out)
+}
+
+// CordovaVersion returns cordova version
+func CordovaVersion() (*ver.Version, error) {
+	cmd := command.New("cordova", "-v")
+	out, err := cmd.RunAndReturnTrimmedCombinedOutput()
+	if err != nil {
+		return nil, err
+	}
+	out = strings.Split(out, "(")[0]
+	out = strings.TrimSpace(out)
+
+	version, err := ver.NewVersion(out)
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to parse version: %s", out)
+	}
+	return version, nil
+}
+
+// LoginCommand runs ionic login comand
+func LoginCommand(username string, password string) error {
```

Comment:
mostly we do not log in packages, to do not influence the client's logs + let the client decide how to run the command models
please reorganize these functions to return a command model / array of command models and solve the logging where you use this functions